smoke_tests/tests/sh/test_endpoint.sh

```diff
@@ -2,22 +2,30 @@
 
 set -e
 
-env="smoke_test_environment"
-
-conf_dir="$(eval echo ~"/.globus_compute/smoke_test")"
-
 cleanup(){
     [[ -d $conf_dir ]] && rm -rf "$conf_dir"
+    [[ -d $GC_HOME ]] && rm -rf "$GC_HOME"
     pkill -P $$
 }
 
+trap cleanup EXIT
+
 (sleep 10s; echo >&2 "Test took too long; killing."; cleanup; pkill -9 -g 0) &
 
-trap cleanup EXIT
+GC_HOME="$(mktemp -d)"
+if [[ $GC_HOME = $HOME ]]; then
+    echo "mktemp failed!"
```

Comment:
mktemp failed is a little cryptic.  Is this to check if already in tmp directory?

Reply:
Because the EXIT handler removes the directory (`rm -rf`), this check is to ensure that if, for whatever reason, `$GC_HOME` is equivalent to the original `$HOME`, that we don't install the trap and don't mistakenly remove someone's home directory.

However, this is me being a bit OCD, because `mktemp`, per it's whole raison d'être, will make a new, heretofore unknown, _randomly named_, directory, and will do it securely.  If it isn't able to create the directory, it will exit with a non-zero exit code, thus triggering `set -e`.

Just consider me paranoid.  :stuck_out_tongue:

Reply:
Slightly changed the text, but mostly added a comment.